ccl/changefeedccl: TestAlterChangefeedAlterTableName failed

ccl/changefeedccl.TestAlterChangefeedAlterTableName [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19921768?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19921768?buildTab=artifacts#/) on master @ [a0362142bb1903793a7a9327c4c5ac8011aa3dbc](https://github.com/cockroachdb/cockroach/commits/a0362142bb1903793a7a9327c4c5ac8011aa3dbc):


```
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a152b17b8fdc5b459db71a1944ab6ad5/logTestAlterChangefeedAlterTableName1164033975
    test_log_scope.go:76: use -show-logs to present logs inline
    alter_changefeed_test.go:1282: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a152b17b8fdc5b459db71a1944ab6ad5/logTestAlterChangefeedAlterTableName1164033975
--- FAIL: TestAlterChangefeedAlterTableName (9.70s)
=== RUN   TestAlterChangefeedAlterTableName/webhook
    helpers_test.go:1235: making server as secondary tenant
    helpers_test.go:1314: making webhook feed factory
    helpers_test.go:1100: forcing enriched envelope for *changefeedccl.webhookFeedFactory - EXPERIMENTAL CHANGEFEED FOR TABLE movr.users WITH OPTIONS (diff, resolved = '100ms', envelope = 'enriched', enriched_properties = 'source')
    alter_changefeed_test.go:1276: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:273
        	            				pkg/ccl/changefeedccl/helpers_test.go:447
        	            				pkg/ccl/changefeedccl/alter_changefeed_test.go:1276
        	            				pkg/ccl/changefeedccl/helpers_test.go:1486
        	            				pkg/ccl/changefeedccl/helpers_test.go:1521
        	Error:      	Received unexpected error:
        	            	expected
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBaseErr
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:415
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:276
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:440
        	            	  | github.com/cockroachdb/cockroach/pkg/util/timeutil.RunWithTimeout
        	            	  | 	pkg/util/timeutil/timeout.go:28
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:436
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:274
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloads
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:447
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestAlterChangefeedAlterTableName.func1
        	            	  | 	pkg/ccl/changefeedccl/alter_changefeed_test.go:1276
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamed.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1486
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamedWithSystem.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1521
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1223
        	            	Wraps: (2) expected
        	            	  |   users: [2]->{"after": {"id": 2, "name": "Bob"}}
        	            	  | got
        	            	  |   riders: [2]->{"after": {"id": 2, "name": "Bob"}}
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestAlterChangefeedAlterTableName/webhook
--- FAIL: TestAlterChangefeedAlterTableName/webhook (9.69s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterChangefeedAlterTableName.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestAlterChangefeedAlterTableName [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19950896?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19950896?buildTab=artifacts#/) on master @ [67cb06a1b702789469ccbccf8368d89790e7db03](https://github.com/cockroachdb/cockroach/commits/67cb06a1b702789469ccbccf8368d89790e7db03):


```
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a152b17b8fdc5b459db71a1944ab6ad5/logTestAlterChangefeedAlterTableName2883254586
    test_log_scope.go:76: use -show-logs to present logs inline
    alter_changefeed_test.go:1289: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a152b17b8fdc5b459db71a1944ab6ad5/logTestAlterChangefeedAlterTableName2883254586
--- FAIL: TestAlterChangefeedAlterTableName (6.53s)
=== RUN   TestAlterChangefeedAlterTableName/webhook
    helpers_test.go:1235: making server as secondary tenant
    helpers_test.go:1314: making webhook feed factory
    helpers_test.go:1100: forcing enriched envelope for *changefeedccl.webhookFeedFactory - EXPERIMENTAL CHANGEFEED FOR TABLE movr.users WITH OPTIONS (diff, resolved = '100ms', envelope = 'enriched', enriched_properties = 'source')
    alter_changefeed_test.go:1283: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:273
        	            				pkg/ccl/changefeedccl/helpers_test.go:447
        	            				pkg/ccl/changefeedccl/alter_changefeed_test.go:1283
        	            				pkg/ccl/changefeedccl/helpers_test.go:1486
        	            				pkg/ccl/changefeedccl/helpers_test.go:1521
        	Error:      	Received unexpected error:
        	            	expected
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBaseErr
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:415
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:276
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:440
        	            	  | github.com/cockroachdb/cockroach/pkg/util/timeutil.RunWithTimeout
        	            	  | 	pkg/util/timeutil/timeout.go:28
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:436
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:274
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloads
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:447
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestAlterChangefeedAlterTableName.func1
        	            	  | 	pkg/ccl/changefeedccl/alter_changefeed_test.go:1283
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamed.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1486
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamedWithSystem.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1521
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1223
        	            	Wraps: (2) expected
        	            	  |   users: [2]->{"after": {"id": 2, "name": "Bob"}}
        	            	  | got
        	            	  |   riders: [2]->{"after": {"id": 2, "name": "Bob"}}
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestAlterChangefeedAlterTableName/webhook
--- FAIL: TestAlterChangefeedAlterTableName/webhook (6.51s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterChangefeedAlterTableName.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestAlterChangefeedAlterTableName [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19966782?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19966782?buildTab=artifacts#/) on master @ [1fc88305c6294dddb54abd1ac3e6e5046edd651f](https://github.com/cockroachdb/cockroach/commits/1fc88305c6294dddb54abd1ac3e6e5046edd651f):


```
=== RUN   TestAlterChangefeedAlterTableName
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a152b17b8fdc5b459db71a1944ab6ad5/logTestAlterChangefeedAlterTableName309627123
    test_log_scope.go:76: use -show-logs to present logs inline
    alter_changefeed_test.go:1289: -- test log scope end --
--- FAIL: TestAlterChangefeedAlterTableName (9.46s)
=== RUN   TestAlterChangefeedAlterTableName/webhook
    helpers_test.go:1238: making server as system tenant
    helpers_test.go:1314: making webhook feed factory
    helpers_test.go:1100: forcing enriched envelope for *changefeedccl.webhookFeedFactory - EXPERIMENTAL CHANGEFEED FOR TABLE movr.users WITH OPTIONS (diff, resolved = '100ms', envelope = 'enriched', enriched_properties = 'source')
    alter_changefeed_test.go:1283: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:273
        	            				pkg/ccl/changefeedccl/helpers_test.go:447
        	            				pkg/ccl/changefeedccl/alter_changefeed_test.go:1283
        	            				pkg/ccl/changefeedccl/helpers_test.go:1486
        	            				pkg/ccl/changefeedccl/helpers_test.go:1521
        	Error:      	Received unexpected error:
        	            	expected
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBaseErr
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:415
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:276
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:440
        	            	  | github.com/cockroachdb/cockroach/pkg/util/timeutil.RunWithTimeout
        	            	  | 	pkg/util/timeutil/timeout.go:28
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:436
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:274
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloads
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:447
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestAlterChangefeedAlterTableName.func1
        	            	  | 	pkg/ccl/changefeedccl/alter_changefeed_test.go:1283
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamed.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1486
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamedWithSystem.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1521
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1223
        	            	Wraps: (2) expected
        	            	  |   users: [2]->{"after": {"id": 2, "name": "Bob"}}
        	            	  | got
        	            	  |   riders: [2]->{"after": {"id": 2, "name": "Bob"}}
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestAlterChangefeedAlterTableName/webhook
--- FAIL: TestAlterChangefeedAlterTableName/webhook (9.44s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterChangefeedAlterTableName.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/alter_changefeed_test.go
+++ b/pkg/ccl/changefeedccl/alter_changefeed_test.go
@@ -1242,7 +1242,17 @@ func TestAlterChangefeedAlterTableName(t *testing.T) {
 		sqlDB.Exec(t,
 			`INSERT INTO movr.users VALUES (1, 'Alice')`,
 		)
-		testFeed := feed(t, f, `CREATE CHANGEFEED FOR movr.users WITH diff, resolved = '100ms'`)
+
+		// TODO(#145927): currently the metamorphic enriched envelope system for
+		// webhook uses source.table_name as the topic. This test expects the
+		// topic name to be durable across table renames, which is not expected
+		// to be true for source.table_name.
+		var args []any
+		if _, ok := f.(*webhookFeedFactory); ok {
+			args = append(args, optOutOfMetamorphicEnrichedEnvelope{reason: "see comment"})
+		}
+
+		testFeed := feed(t, f, `CREATE CHANGEFEED FOR movr.users WITH diff, resolved = '100ms'`, args...)
 		defer closeFeed(t, testFeed)
 
 		assertPayloads(t, testFeed, []string{

